src/data_access/FirebaseUserDataAccessObject.java

```diff
@@ -17,22 +21,93 @@ public class FirebaseUserDataAccessObject
   static OkHttpClient client = new OkHttpClient();
 
   @Override
-  public User get(String email) {
-    Request request =
-        new Request.Builder()
-            .url(
-                Constants.FIREBASE_URL
-                    + "users/"
-                    + email
-                    + ".json") // TODO: Ensure this successfully searches by email
-            .method("GET", null)
-            .build();
+  public Optional<User> get(String email, String password) {
+    // Authentication Request
+    JSONObject jsonBody = new JSONObject();
+    jsonBody.put("email", email);
+    jsonBody.put("password", password);
+    jsonBody.put("returnSecureToken", true);
+
+    MediaType JSON = MediaType.parse("application/json; charset=utf-8");
+    RequestBody body = RequestBody.create(JSON, jsonBody.toString());
+
+    String authUrl =
+        "https://identitytoolkit.googleapis.com/v1/accounts:signInWithPassword?key=AIzaSyBuJk14Gljk-chdN_9YVywSKnf38ttwUVg";
+
+    Request authRequest = new Request.Builder().url(authUrl).post(body).build();
+
+    try {
+      Response authResponse = client.newCall(authRequest).execute();
+      String authResponseData = authResponse.body().string();
+
+      if (!authResponse.isSuccessful()) {
+        System.out.println("Authentication failed: " + authResponseData);
+        return Optional.empty();
```

Comment:
We need to add error handling to this in the future.
Consider throwing an error and catching it in the LoginInteractor.
When this is implemented, we should change the `Optional<User>` to a `User`.

Reply:
Yes, I agree. Will address this in a future PR dedicated to error handling for auth.

Reply:
Creating issue #33 to track this.